src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/References/ReferenceCleanupService.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Exceptions;
+using Microsoft.VisualStudio.ProjectSystem.VS.References.Roslyn;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.References
+{
+    [Export(typeof(IProjectSystemReferenceCleanupService))]
+    internal partial class ReferenceCleanupService : IProjectSystemReferenceCleanupService
+    {
+        private readonly Dictionary<ProjectSystemReferenceType, AbstractReferenceHandler> _mapReferenceTypeToHandler;
+
+        private readonly IProjectServiceAccessor _projectServiceAccessor;
+
+        [ImportingConstructor]
+        public ReferenceCleanupService(IProjectServiceAccessor projectServiceAccessor)
+        {
+            _projectServiceAccessor = projectServiceAccessor;
+
+            _mapReferenceTypeToHandler =
+                new Dictionary<ProjectSystemReferenceType, AbstractReferenceHandler>()
+                {
+                    { ProjectSystemReferenceType.Project, new ProjectReferenceHandler() },
+                    { ProjectSystemReferenceType.Package, new PackageReferenceHandler() },
+                    { ProjectSystemReferenceType.Assembly, new AssemblyReferenceHandler()},
+                    { ProjectSystemReferenceType.Unknown, null }
+                };
+        }
+
+        public async Task<ImmutableArray<ProjectSystemReferenceInfo>> GetProjectReferencesAsync(string projectPath, string targetFrameworkMoniker, CancellationToken cancellationToken)
+        {
```

Comment:
This takes a TFM but you ignore it. What does this TFM represent?

Reply:
this will be removed from the interface  https://github.com/dotnet/roslyn/pull/48990